opencog/atoms/pattern/PatternLink.h

```diff
@@ -174,7 +174,7 @@ class PatternLink : public PrenexLink
 	const HandleSeq& get_fixed(void) const { return _fixed; }
 	const HandleSeq& get_virtual(void) const { return _virtual; }
 
-	bool satisfy(PatternMatchCallback&) const;
+	bool satisfy(PatternMatchCallback&, AtomSpace* as);
```

Comment:
Actually, I'm not confident about the order of arguments. Should `as` come first or second?

@linas I know that you've commented about argument orders for instance in https://github.com/opencog/atomspace/blob/master/opencog/scm/opencog/base/utilities.scm#L1177 but I don't understand the rationals.

The only rational I have for argument order is from Haskell, put the most frequently changing arguments last, because that way you can simplify a function call by partially calling it on the first arguments (function currying).